File: src/types/bible.ts
export interface BibleBook {
  id: string;
  name: string;
  chapters: number;
  genre: BibleGenre;
  period?: string;
  author?: string;
}

export type BibleGenre = 'Law' | 'History' | 'Wisdom' | 'Prophets' | 'Gospels' | 'Letters';

export interface HistoricalContext {
  period: string;
  date: string;
  background: string;
  concurrent: string[];
}

export interface GeographicContext {
  mainLocation: string;
  relatedSites: string[];
  routes: string[];
}

export interface ChapterIntroduction {
  summary: string;
  themes: string[];
  characters: {
    name: string;
    role: string;
  }[];
  structure: string[];
}

export interface BibleViewState {
  selectedView: 'traditional' | 'chronological';
  showVerseNumbers: boolean;
  expandedSection: string | null;
}

export interface ReadingProgress {
  currentBook: BibleBook;
  currentChapter: number;
  overallProgress: number;
}

File: backend/app/models/bible.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional

class BibleBook(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: str
    name: str
    chapters: int
    genre: str
    period: Optional[str] = None
    author: Optional[str] = None

class HistoricalContext(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    period: str
    date: str
    background: str
    concurrent: List[str]

class GeographicContext(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    mainLocation: str
    relatedSites: List[str]
    routes: List[str]

class Character(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    name: str
    role: str

class ChapterIntroduction(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    summary: str
    themes: List[str]
    characters: List[Character]
    structure: List[str]

class Verse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    number: int
    text: str
    explanation: Optional[str] = None

class ReadingProgress(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    currentBook: BibleBook
    currentChapter: int
    overallProgress: int